src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -2,43 +2,678 @@
+            }
+        }
+
+        [CLSCompliant(false)]
+        public static unsafe Vector128<ulong> Create(ulong e0, ulong e1)
+        {
+            if (Sse2.IsSupported)
+            {
+                // TODO-CQ: Use the CreateScalar which leaves upper bits uninitialized
+
+                return Sse2.UnpackLow(CreateScalar(e0), CreateScalar(e1));  // { e0, e1 }
+            }
+            else
+            {
+                var pResult = stackalloc ulong[2]
+                {
+                    e0,
+                    e1,
+                };
+
+                return Unsafe.AsRef<Vector128<ulong>>(pResult);
+            }
+        }
+
+        public static unsafe Vector128<T> Create<T>(Vector64<T> lower, Vector64<T> upper) where T : struct
+        {
+            var result = Vector128<T>.Zero; // Throws PNSE for unsupported T
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector128<T>, byte>(ref result), lower);
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector128<T>, byte>(ref Unsafe.AddByteOffset(ref result, 8)), upper);
+            return result;
+        }
+
+        public static unsafe Vector128<byte> CreateScalar(byte value)
+        {
+            if (Sse2.IsSupported)
+            {
+                return Sse2.ConvertScalarToVector128UInt32(value).AsByte();
+            }
+            else
+            {
+                var result = Vector128<byte>.Zero;
+                Unsafe.WriteUnaligned(ref Unsafe.As<Vector128<byte>, byte>(ref result), value);
+                return result;
+            }
+        }
+
+        public static unsafe Vector128<double> CreateScalar(double value)
+        {
+            // TODO-CQ: Use the CreateScalar which leaves upper bits uninitialized
+
+            var result = Vector128<double>.Zero;
+            Unsafe.WriteUnaligned(ref Unsafe.As<Vector128<double>, byte>(ref result), value);
+            return result;
+        }
+
+        public static unsafe Vector128<short> CreateScalar(short value)
+        {
+            if (Sse2.IsSupported)
+            {
+                return Sse2.ConvertScalarToVector128Int32(value).AsInt16();
+            }
+            else
+            {
+                var result = Vector128<short>.Zero;
+                Unsafe.WriteUnaligned(ref Unsafe.As<Vector128<short>, byte>(ref result), value);
+                return result;
+            }
+        }
+
+        public static unsafe Vector128<int> CreateScalar(int value)
+        {
+            if (Sse2.IsSupported)
+            {
+                return Sse2.ConvertScalarToVector128Int32(value);
+            }
+            else
+            {
+                var result = Vector128<int>.Zero;
+                Unsafe.WriteUnaligned(ref Unsafe.As<Vector128<int>, byte>(ref result), value);
+                return result;
+            }
+        }
+
+        public static unsafe Vector128<long> CreateScalar(long value)
+        {
+            if (Sse2.IsSupported)
```

Comment:
Currently, we need `Environment.Is64BitProcess` check for the SSE2 implementation of `CreateScalar(long/ulong value)`.